src/chain.cpp

```diff
@@ -5,6 +5,11 @@
 
 #include <chain.h>
 #include <util/time.h>
+#include <validation.h>
```

Comment:
This is a circular dependency. 
```
"chain.h -> validation.h -> chain.h" 
```

I'm not quite sure how to fix it -- we can't simply forward declare `ChainManager` here since the line below requires knowledge of the fields of `ChainManager`:
```
m_pcontext->chainman->m_blockman.m_dirty_blockindex.insert(this);
 ```